From 84dfe4d813257a3a4de0b44648d7b6a8b330dcb0 Mon Sep 17 00:00:00 2001
From: Aaradhana Sahu <quic_aarasahu@quicinc.com>
Date: Wed, 25 Sep 2024 11:48:46 +0530
Subject: [PATCH] wifi: ath12k: set rxmon_disable flag during monitor vap down

Currently rxmon_diable flag is false during monitor vap down, due
to this monitor config is still enable for hardware.

Setting rxmon_diable flag true when monitor interface goes
down, so hardware can't perform any write operation.

Signed-off-by: Aaradhana Sahu <quic_aarasahu@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_tx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp_tx.c b/drivers/net/wireless/ath/ath12k/dp_tx.c
index 3cc9ff0..dec6dcc 100644
--- a/drivers/net/wireless/ath/ath12k/dp_tx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_tx.c
@@ -2246,6 +2246,8 @@ int ath12k_dp_tx_htt_rx_monitor_mode_ring_config(struct ath12k *ar, bool reset)
 		tlv_filter.info1 |=
 			FIELD_PREP(HTT_RX_TLV_FILTER_INFO1_RX_PKT_TLV_OFFSET,
 				   HTT_RX_RING_PKT_TLV_OFFSET);
+	} else {
+		tlv_filter.rxmon_disable = true;
 	}
 
 	for (i = 0; i < ar->ab->hw_params->num_rxmda_per_pdev; i++) {
-- 
2.34.1

